src/binding/petsc4py/src/petsc4py/PETSc/DMDA.pyx

```diff
@@ -353,44 +786,168 @@ cdef class DMDA(DM):
                                           _ymin, _ymax,
                                           _zmin, _zmax) )
 
-    def setCoordinateName(self, index, name):
+    def setCoordinateName(self, index: int, name: str) -> None:
+        """Set the name of the coordinate direction.
+
+        Logically collective; ``name`` must contain a common value.
+
+        Parameters
+        ----------
+        index
+            The coordinate number for the DMDA (``0``, ``1``, ..., ``dim-1``).
+        name
+            The name of the coordinate.
+
+        See Also
+        --------
+        getCoordinateName, petsc.DMDASetCoordinateName
+
+        """
         cdef PetscInt ival = asInt(index)
         cdef const char *cval = NULL
         name = str2bytes(name, &cval)
         CHKERR( DMDASetCoordinateName(self.dm, ival, cval) )
 
-    def getCoordinateName(self, index):
+    def getCoordinateName(self, index: int) -> str:
+        """Return the name of a coordinate direction.
+
+        Not collective; the returned value will contain a common value.
+
+        Parameters
+        ----------
+        index
+            The coordinate number for the DMDA (``0``, ``1``, ..., ``dim-1``).
+
+        See Also
+        --------
+        setCoordinateName, petsc.DMDAGetCoordinateName
+
+        """
         cdef PetscInt ival = asInt(index)
         cdef const char *cval = NULL
         CHKERR( DMDAGetCoordinateName(self.dm, ival, &cval) )
         return bytes2str(cval)
 
     #
 
-    def createNaturalVec(self):
+    def createNaturalVec(self) -> Vec:
+        """Create a vector that will hold values in the natural numbering.
+
+        The number of local entries in the vector on each process is the same
+        as in a vector created with `DM.createGlobalVec`.
+
+        Collective.
+
+        See Also
+        --------
+        petsc.DMDACreateNaturalVector
+
+        """
         cdef Vec vn = Vec()
         CHKERR( DMDACreateNaturalVector(self.dm, &vn.vec) )
         return vn
 
-    def globalToNatural(self, Vec vg, Vec vn, addv=None):
+    def globalToNatural(
+        self,
+        Vec vg,
+        Vec vn,
+        addv: InsertMode | None = None,
+    ) -> None:
+        """Map values to the "natural" grid ordering.
+
+        Output Parameter
+        l - the natural ordering values
+        l - the natural ordering values
+        l - the global values in the natural ordering
+
+        Notes
+        The global and natural vectors used here need not be the same as those obtained from DMCreateGlobalVector() and DMDACreateNaturalVector(), BUT they must have the same parallel data layout; they could, for example, be obtained with VecDuplicate() from the DMDA originating vectors.
+
+        You must call DMDACreateNaturalVector() before using this routine
+
+        Neighbor-wise collective.
```

Comment:
Collective or others go after the first line. Check in other places.

Reply:
That's another one for the mass search&replace. I got it this way everywhere.